packages/beacon-state-transition/src/allForks/util/epochProcess.ts

```diff
@@ -118,8 +121,13 @@ export function prepareEpochProcessState<T extends allForks.BeaconState>(state:
       activeCount += 1;
     }
 
-    if (v.exitEpoch !== FAR_FUTURE_EPOCH && v.exitEpoch > exitQueueEnd) {
-      exitQueueEnd = v.exitEpoch;
+    if (v.exitEpoch !== FAR_FUTURE_EPOCH) {
+      if (v.exitEpoch > exitQueueEnd) {
+        exitQueueEnd = v.exitEpoch;
+        exitQueueEndChurn = 1;
```

Comment:
Is this right?
exitQueueEndChurn will be reset to 1 in this case?

Reply:
yeah it should not be zero. `exitQueueEndChurn` means number of `max(exitEpoch)` so if we find a new higher `exitEpoch`, we found the 1st validator with that highest `exitEpoch`

tbh I made it start from 0 and then spec tests failed, so I fixed it this way :D

Reply:
Oh, I see

Before:
- Loop thru validators, find the max exit epoch that isn't `FAR_FUTURE_EPOCH`, set that to `exitQueueEnd`.
- Loop thru validators again, count all validators with exit epoch equal to `exitQueueEnd`, set that to `exitQueueEndChurn`

Now:
- Loop once thru validators, start counting validators with the largest exit epoch (that isn't `FAR_FUTURE_EPOCH`), reset the count and largest exit epoch if a larger exit epoch is found